lib/internal/modules/cjs/loader.js

```diff
@@ -664,14 +664,8 @@ Module.prototype._compile = function(content, filename) {
 
   content = stripShebang(content);
 
-  // create wrapper function
-  var wrapper = Module.wrap(content);
-
-  var compiledWrapper = vm.runInThisContext(wrapper, {
-    filename: filename,
-    lineOffset: 0,
-    displayErrors: true
-  });
+  const func = vm.getWrappedFunction(content, false);
+  var compiledWrapped = func(); // FIXME(ryzokuken): This won't work.
```

Comment:
I think you don’t want to call `func` yet, but instead have `compiledWrapper = vm.getWrappedFunction(…)` and leave the rest?

Reply:
originally, `compilerWrapper` would return a `v8::Value` that'd be returned as the result of your script, which it'll receive by calling `script->Run()`.

All that I think needs to be done here is somehow calling the function (`func->Call()`) with the same context (`env->context()`). All the needed vars (`require`, `__dirname`, etc) will be in there, right?

Reply:
@addaleax on second thought, you're probably right.